src/engine/Server.cpp

```diff
@@ -28,36 +28,30 @@ Server::Server(unsigned short port, size_t numThreads,
     : numThreads_(numThreads),
       port_(port),
       accessToken_(std::move(accessToken)),
-      allocator_{ad_utility::makeAllocationMemoryLeftThreadsafeObject(
-                     maxMem.getBytes()),
-                 [this](size_t numBytesToAllocate) {
-                   cache_.makeRoomAsMuchAsPossible(MAKE_ROOM_SLACK_FACTOR *
-                                                   numBytesToAllocate /
-                                                   sizeof(Id));
-                 }},
+      allocator_{
+          ad_utility::makeAllocationMemoryLeftThreadsafeObject(
+              maxMem.getBytes()),
+          [this](size_t numBytesToAllocate) {
+            cache_.makeRoomAsMuchAsPossible(ad_utility::MemorySize::bytes(
+                MAKE_ROOM_SLACK_FACTOR * numBytesToAllocate / sizeof(Id)));
+          }},
       index_{allocator_},
       enablePatternTrick_(usePatternTrick),
       // The number of server threads currently also is the number of queries
       // that can be processed simultaneously.
       threadPool_{numThreads} {
-  // TODO<joka921> Write a strong type for KB, MB, GB etc and use it
-  // in the cache and the memory limit
-  // Convert a number of gigabytes to the number of Ids that find in that
-  // amount of memory.
-  auto toNumIds = [](size_t gigabytes) -> size_t {
-    return gigabytes * (1ull << 30u) / sizeof(Id);
-  };
   // This also directly triggers the update functions and propagates the
   // values of the parameters to the cache.
   RuntimeParameters().setOnUpdateAction<"cache-max-num-entries">(
       [this](size_t newValue) { cache_.setMaxNumEntries(newValue); });
   RuntimeParameters().setOnUpdateAction<"cache-max-size-gb">(
-      [this, toNumIds](size_t newValue) {
-        cache_.setMaxSize(toNumIds(newValue));
```

Comment:
Will that be a separate PR that converts the `RuntimeParameters`?

Reply:
Yes. Otherwise, this PR will just lay around for 1+ month again.